.github/workflows/pr-validation.yml

```
@@ -1,51 +1,58 @@
-# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
-# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
-
-name: Build Validation
-
-on:
-  push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
-  merge_group:
-
-jobs:
-  build:
-
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
-
-    steps:
-    - uses: actions/checkout@v4
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install flake8 pytest
-        pip install -r requirements.txt
-    - name: Lint with flake8
-      run: |
-        flake8 . --count --show-source --statistics --exit-zero
-    - name: Pytest unit tests
-      run: |
-        pytest -m "not e2e" --verbose
-
-    # Sidecar for running e2e tests requires Go SDK
-    - name: Install Go SDK
-      uses: actions/setup-go@v5
-      with:
-        go-version: 'stable'
-
-    # Install and run the durabletask-go sidecar for running e2e tests
-    - name: Pytest e2e tests
-      run: |
-        go install github.com/microsoft/durabletask-go@main
-        durabletask-go --port 4001 &
-        pytest -m "e2e" --verbose
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Build Validation
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  merge_group:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install durabletask dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        pip install -r requirements.txt
+    - name: Install durabletask-azuremanaged dependencies
+      working-directory: examples/dts
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Lint with flake8
+      run: |
+        flake8 . --count --show-source --statistics --exit-zero

```

```suggestion
        flake8 . --count --show-source --statistics

```

We want this step to fail if there are linter issues, so let's not use `--exit-zero`

---

I see that was there before - but let's actually change this!

---

This leads to a lot of lint errors that were being ignored. I'll create an additional PR afterwards that removes this flag and addresses the linting problems so I don't clutter this PR too much

---

Ok, please do that! Just because existing linter errors exist doesn't mean we should leave them in place.